mypy/messages.py

```diff
@@ -136,11 +137,13 @@ def add_errors(self, messages: 'MessageBuilder') -> None:
                 for info in errs:
                     self.errors.add_error_info(info)
 
-    def disable_errors(self) -> None:
+    @contextmanager
+    def disable_errors(self) -> ContextManager[None]:
```

Comment:
Return type should be `Iterable[None]` (or something with `Generator`). The function returns an iterable, the decorator turns it into a ContextManager.

Reply:
Oops, thanks. I guess I was trying to placate PyCharm which complained that it doesn't have `__enter__()` when I tried to use it "manually" within `analyze_type_type_member_access`.

Maybe I should just bite the bullet and use `with` in `analyze_type_type_member_access` (and indent all the code).